File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-1.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        ans=[-1]*len(arr)
        i=len(arr)-1
        great=-1
        for num in reversed(arr):
            arr[i]=great    
            great=max(great,num)
           
            i-=1

        return arr      

File: Data Structures & Algorithms/counting-bits/submission-1.cpp
class Solution {
public:
    vector<int> countBits(int n) {
         
         vector<int>arr(n+1,0);
         int offset=1;
         for(int i=1;i<=n;i++){
            
            if(offset*2==i){
                offset=i;
            }
            arr[i]=1+arr[i-offset];




         }
        return arr;
    }
};
